### ISSUE TITLE
Add prestissimo for composer speed up (advice)

### ISSUE THREAD

**Issue by plispe** [heroku/heroku-buildpack-php#143]
Hello, what do you think about[ Prestissimo](https://github.com/hirak/prestissimo)? I thing it would be great to have this out of the box in the build pack.

**Comment by dzuelke** [comment]
Yes, experiments with that are on my list :)

I wouldn't expect the performance gains to be *too* massive though; user dependencies after the first push get cached anyway. For platform installs (PHP, extensions, etc), the issue is that the cache is forcefully disabled, since otherwise the Heroku build cache would grow too large, and caching has no benefit, since both the cache and the platform packages are hosted on S3 - the additional overhead of a couple of HTTPS requests to fetch individual tarballs is negligible (I've benchmarked that before).

Of course the parallelism will be quite beneficial! In a few weeks, I'll put some focus on smaller tarballs and other performance enhancements, and I'm sure I can shave quite a few seconds off every deploy.

I'd rather see the whole thing get merged into Composer though; I already talked about it with @naderman.

**Comment by dzuelke** [comment]
The other question of course is... if this is installed on Heroku as a global plugin, then will it conflict with anything else? Do users use custom plugins with installer hooks that are maybe incompatible? How do people then disable it? Can you tell them that this might be the issue on build fail, or will they be lost with no clue what's happening (because a local `composer install` works)?

**Comment by plispe** [comment]
Thank you for your response and your point of view. I think that, the best solution would be to merge prestissimo with composer :-)

**Comment by dzuelke** [comment]
Yeah; that would probably also significantly reduce the codebase, and prevent compatibility issues like hirak/prestissimo#40

**Comment by holtkamp** [comment]
@dzuelke still considering adopting prestissimo for improved performance of composer installation?

Regarding https://github.com/heroku/heroku-buildpack-php/issues/148, installing all dependencies from the `composer.lock` file (all from cache) still takes quite some time on Heroku. I was hoping for a performance improvement when applying parallelism...

**Comment by dzuelke** [comment]
Thanks for reminding me to close this ;) Composer 2 will have parallel download support, and Prestissimo anyway only does the downloading in parallel, not the extraction...
